lib/extract/netcdf/kernel_data_netcdf.f90

```diff
@@ -0,0 +1,278 @@
+module psy_data_mod
+
+    implicit none
+
+    ! Contains functionality to write and read variables
+    type, public:: PSyDataType
+        integer, private                   :: ncid
+        integer, dimension(:), allocatable :: var_id
+        integer                            :: num_pre_vars, &
+                                              num_post_vars
+        integer                            :: next_var_index
+    contains
+        procedure :: DeclareScalarInteger, WriteScalarInteger
+        procedure :: DeclareScalarReal,    WriteScalarReal
+        procedure :: DeclareFieldDouble,   WriteFieldDouble
+        procedure :: ReadScalarInteger, ReadScalarReal, ReadFieldDouble
+        procedure :: PreStart, PreEndDeclaration, PreEnd
+        procedure :: PostStart, PostEnd
+        procedure :: OpenRead
+
+        generic, public :: PreDeclareVariable => DeclareScalarInteger, &
+                                                 DeclareScalarReal,    &
+                                                 DeclareFieldDouble
+
+        generic, public :: ProvideVariable => WriteScalarInteger, &
+                                              WriteScalarReal,    &
+                                              WriteFieldDouble
+
+        generic, public :: ReadVariable => ReadScalarInteger, &
+                                           ReadScalarReal,    &
+                                           ReadFieldDouble 
+    end type PSyDataType
+
+Contains
+
+    ! -------------------------------------------------------------------------
+    ! Checks if the return value from a netcdf call indicates an error.
+    ! If so, print the corresponding error message.
+    function CheckError(retval) 
+        use netcdf
+        implicit none
+        integer :: CheckError
+        integer, intent(in) :: retval
+        if (retval /= nf90_noerr) then
+            print *,"NetCDF Error:"
+            print *,trim(nf90_strerror(retval))
+        endif
+        CheckError = retval
+    end function CheckError
+
+    ! -------------------------------------------------------------------------
+    subroutine PreStart(this, module_name, kernel_name, num_pre_vars, &
+                        num_post_vars)
+        use netcdf, only : nf90_create, NF90_CLOBBER
+        implicit none
+        class(PSyDataType), intent(inout) :: this
+        character(*), intent(in) :: module_name, kernel_name
+        integer, intent(in)      :: num_pre_vars, num_post_vars
+        integer :: retval
+
+        retval = CheckError(nf90_create(module_name//"-"//kernel_name//".nc", &
+                                        NF90_CLOBBER, this%ncid))
```

Comment:
Please abort if the file creation fails (or take some appropriate action).

Reply:
I added an abort in case of any error - if anything goes wrong with a netcdf operation, things will typically go completely wrong (e.g. variable definition incorrect, later you can't write that variable). I think it is best to abort early.